### DISCUSSION TITLE
Namespace in Monolingual base language file

### DISCUSSION THREAD

**Discussion by alikleit** [WeblateOrg/weblate#8613] [Q&A] [answered]
Is there a file mask for the "Monolingual base language file" that we can use to point to multiple base files foreach locale? for example, I have the following structure for the base files in one application:

/locales/helpers/en.json
/locales/users/en.json
/locales/products/en.json
and so on... 

and I have the file mask as:
/locales/**/*.json

the "**" represents the namespace in i18next which divides locales into groups, which is efficient to provide more readable and clean locales for the developers.

For the "Monolingual base language file " I can currently define only one source file, for example:
/locales/helpers/en.json

which will take only the 'helpers' namespace, I did some research and I saw the only to include the namespace is to define a component foreach namespace under a project which is quite repetitive and time consuming to manage properly. as you have to define the same configuration  but with different "Monolingual base language file " path.

Is there a  way to define a namespace in the "Monolingual base language file" something like " /locales/**/en.json " or " /locales/{{ns}}/en.json ", if so is there a proper example on the configuration you can provide?

**Comment by nijel** [comment, marked as answer]
Create a component for the namespace that is most likely to stay forever, and then set up [the discovery add-on](https://docs.weblate.org/en/latest/admin/addons.html#component-discovery) to create others.
